### PR TITLE
atom: update to version 1.36

### PR DESCRIPTION
**Pull request by lierdakil**
Please fill in this template.

- [X] Use a meaningful title for the pull request. Include the name of the package modified.
- [X] Test the change in your own code. (Compile and run.)
- [X] Add or edit tests to reflect the change. (Run with `npm test`.)
- [X] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [X] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [X] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If changing an existing definition:
- [X] Provide a URL to documentation or source code which provides context for the suggested changes: https://atom.io/docs/api/v1.36.0
- [X] Increase the version number in the header if appropriate.
- [ ] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

Long story short, a few updates. Note though that there are a couple things we missed a while ago (pre 1.31). Also, one minor tweak is that `TextEditor.getLastCursor()` can return `undefined` for e.g. readonly editors. Other changes should be self-explanatory.

### CODE REVIEW

**Comment by GlenCFL** [comment]
LGTM
`TextEditor::getLastCursor` being able to return undefined doesn't seem intended, but I think it can after looking through their code (located [here](https://github.com/atom/atom/blob/v1.36.0/src/text-editor.js#L2823)).

**Comment by lierdakil** [comment]
@GlenCFL, `getLastCursor` returning `undefined` was something I've personally run into at some point, but tbh I don't exactly remember under what circumstances, and looking through commit history, this is indeed probably unintended. Perhaps filing a bug report on Atom repo would've been a better option now that I think about it.

**Comment by lierdakil** [comment]
Yep, looking at the code, `getLastCursor` shouldn't return `undefined` ever. Couldn't reproduce it either on 1.36. So my bad. Will whip up an amendment shortly.
